### ISSUE TITLE
Array type model set to null and schema properties set to null not accepted by formly

### ISSUE THREAD

**Issue by Adam-Michalski** [ngx-formly/ngx-formly#1764]
[x] bug report => search github for a similar issue or PR before submitting

**Current behavior**

Model for array set to `null` is not accepted
Properties set to `null` for fields is not accepted

**Expected behavior**

Schema with properties set to `null` validates properly but formly does not accept this and throws error `Error: Cannot read property 'length' of null`

Model set to null is accepted for every type but not for arrays (It should work for arrays also).

 [Instance Data Model](https://json-schema.org/latest/json-schema-core.html#rfc.section.4.2.1)
An empty schema is a JSON Schema with no properties -> [JSON Schema Values and Keywords](https://json-schema.org/latest/json-schema-core.html#rfc.section.4.3.1)

**Minimal reproduction of the problem with instructions**

Here is example for every combination in [StackBlitz](https://stackblitz.com/edit/angular-formly-issues?file=src/assets/json-schema/arrays.json) example:

- [simple schema] properties set to `null` in object
- [nested schema] properties set to `null` in array `items`,
- [arrays schema] model set to `null` for array model,
- [numbers schema] model set to `null` in object model

**What is the motivation / use case for changing the behavior?**

Schema with not specified preferences is oportunity to implement on demand rebuild of formly form based on changes in one of fields specified property for refresh.

Schema does not specify properties for first build cause we don't know what response we would get.

In cause of errors I must map model and schema and add elements like in schema when `properties:null`=>`properties: {}` the same for `items.properties` and in model when `type:'array'` `model:null` => `model:[]` Which I think should be done like in object when model is null.

**Comment by kenisteward** [comment]
@aitboudad billable shouldn't be allowed unless the typenisnset to null! I saw this issue too late but I think that change may need to be reverrted unless you didn't based on having the jsonschema type be null.


From what I saw we currently don't have logic that handles multi type processing

**Comment by Adam-Michalski** [comment]
I am facing this problem right now and I think it can be handled by something like
```
      {
        // This name matches ["array", "null"] JSON schema type
        name: 'array,null',
        extends: JsonSchemaType.string,
        defaultOptions: {
          parsers: [
            // This parser causes empty input to be represented in model as null
            nullIfEmpty,
          ],
        },
      }
```
Or just array

```
      {
        name: 'array',
        extends: JsonSchemaType.string,
        defaultOptions: {
          parsers: [
            // This parser causes empty input to be represented in model as null
            nullIfEmpty,
          ],
        },
      }
```

`nullIfEmpty`
```
export function nullIfEmpty<T>(value: T[]): T[] {
  if (value && value.length) {
    return null;
  }
  return value;
}
```

**Comment by aitboudad** [comment]
@kenisteward I'm aware of that case, it should be covered once adding the `type` validator in the next version (`5.5`).

passing an `["array", "null"]` should be supported in the next fix release `5.4.2`

**Comment by aitboudad** [comment]
released in `5.4.2` version.
